### PR TITLE
assign: fix assigning papers to new author

### PR DESCRIPTION
**Pull request by MJedr**
ref: #2183

### CODE REVIEW

File: backend/tests/integration-async/assign/test_views.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+from helpers.providers.faker import faker
+from helpers.utils import create_user
+from invenio_accounts.testutils import login_user_via_session
+from invenio_db import db
+from sqlalchemy.exc import ResourceClosedError
+
+from inspirehep.records.api import AuthorsRecord, LiteratureRecord
+
+
+def test_assign_doesnt_craise_resources_closed_error(inspire_app, override_config):
```
Comment (michamos) [inline backend/tests/integration-async/assign/test_views.py:11]:
```suggestion
def test_assign_doesnt_raise_resource_closed_error(inspire_app, override_config):
```

File: backend/tests/integration-async/assign/test_views.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+                                paper_2["control_number"],
+                                paper_3["control_number"],
+                            ],
+                            "from_author_recid": author_record["control_number"],
+                        }
+                    ),
+                    content_type="application/json",
+                )
+        except ResourceClosedError:
+            assert False
```
Comment (michamos) [inline backend/tests/integration-async/assign/test_views.py:77]: are you sure this test is failing without your change? I tried something extremely similar yesterday on a worker and I didn't get any exception.

File: backend/inspirehep/assign/views.py [hunk 3]
```diff
@@ -49,7 +49,12 @@ def assign_papers(
 ):
     author_bai = get_values_for_schema(to_author_record["ids"], "INSPIRE BAI")[0]
     for record in author_papers:
-        lit_author = get_author_by_recid(record, from_author_recid)
+        try:
+            lit_author = get_author_by_recid(record, from_author_recid)
+        # it might happen that stub author BAI already existed
+        # and we already assigned papers to his profile.
+        except StopIteration:
+            continue
```
Comment (michamos) [inline backend/inspirehep/assign/views.py:57]: This is a workaround for #2177. I don't think we should ignore the issue but rather find and fix the root cause. How is this related to this issue?

Comment (MJedr) [reply backend/inspirehep/assign/views.py:57]: I am not sure if that's a workaround for another issue cause this one is about `get_author_by_recid` and the other one is about BAI. In the case that I try to fix the following problem, due to which my test for commiting was failing:   
we create a new stub author and [here](https://github.com/inspirehep/inspirehep/blob/39bbacf372b720b6cfed7ce577f74abf72256dc5/backend/inspirehep/records/api/authors.py#L91) we assign reference to this new author so if we try to get an old author by recid it's not valid anymore and we get an error.

Comment (michamos) [reply backend/inspirehep/assign/views.py:57]: You're right of course about the issue being different, I got confused by the `StopIteration` that made it look superficially similar. I don't understand the failure mode however. Let's discuss it.

**Review by michamos — APPROVED** [review]
